ts/features/wallet/onboarding/cobadge/saga/orchestration/addCoBadgeToWallet.ts

```diff
@@ -48,24 +49,34 @@ export function* addCoBadgeToWalletAndActivateBpd() {
     coBadgeWorkUnit
   );
   if (res !== "back") {
-    // If the addition starts from "WALLET_ONBOARDING_COBADGE_CHOOSE_TYPE", remove from stack
+    // If the addition starts from "WALLET_ONBOARDING_COBADGE_CHOOSE_TYPE" or from "WALLET_ADD_CARD", remove from stack
     // This shouldn't happens if all the workflow will use the executeWorkUnit
     const currentRoute: ReturnType<typeof navigationCurrentRouteSelector> = yield select(
       navigationCurrentRouteSelector
     );
 
+    if (
```

Comment:
Could it be a good refactoring?
it avoids to check multiple times if `currentRoute` is some and instead of putting 2 consecutive navigations back it pops 1 and then it does back

```typescript
if (currentRoute.isSome()) {
      yield put(navigationHistoryPop(1));
      yield put(NavigationActions.back());

      if (currentRoute.value === "WALLET_ONBOARDING_COBADGE_CHOOSE_TYPE") {
        yield put(NavigationActions.back());
        const newRoute: ReturnType<typeof navigationCurrentRouteSelector> = yield select(
          navigationCurrentRouteSelector
        );
        if (
          res === "completed" &&
          newRoute.isSome() &&
          newRoute.value === "WALLET_BANCOMAT_DETAIL"
        ) {
          yield put(NavigationActions.back());
        }
      }
    }
```

Reply:
Sounds good. Refactored in [702efbc](https://github.com/pagopa/io-app/pull/2931/commits/702efbc108aa3f11f2fea7f6bfa46301f7f4d300).

The only thing different from your proposal is that I maintain the first che on the `currentRoute`.
```typescript

if (currentRoute.isSome()) {
      if (currentRoute.value === ROUTES.WALLET_ADD_CARD) {
        yield put(navigationHistoryPop(1));
        yield put(NavigationActions.back());
      }
   ... 
```